### PR TITLE
test: replace lfs with luv and vim.fs

### PR DESCRIPTION
**Pull request by dundargoc**
luv already pretty much does everything lfs does, so this duplication
of dependencies isn't needed.

### CODE REVIEW

File: test/helpers.lua [hunk 1]
```diff
@@ -843,6 +843,10 @@ function module.read_nvim_log(logfile, ci_rename)
   return log
 end
 
+function module.mkdir(path)
+  return lfs.mkdir(path)
```
Comment (justinmk) [inline test/helpers.lua:847]: the test runner is nvim, do we have `vim.fn` available? (or a TODO once `nvim -ll` becomes `nvim -l`)?

**Review by justinmk — APPROVED** [review]

**Comment by dundargoc** [comment]
Luv.fs_mkdir

> just hardcode 0755
> 493 in decimal

File: test/helpers.lua (hunk 1 above)
Comment (dundargoc) [reply test/helpers.lua:847]: > test/helpers.lua:847: attempt to index field 'fn' (a nil value)

I dunno

Comment (dundargoc) [reply test/helpers.lua]: We're currently running the test with `-ll`. Unsure what that means.

**Review by lewis6991 — APPROVED** [review]

**Pull request merged**